src/parser/monk/mistweaver/modules/core/HotTrackerMW.js

```diff
@@ -0,0 +1,64 @@
+import SPELLS from 'common/SPELLS';
+import HotTracker from 'parser/shared/modules/HotTracker';
+
+class HotTrackerMW extends HotTracker {
+
+  constructor(...args) {
+    super(...args);
+    this.mistwrapActive = this.selectedCombatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id);
+    this.upwellingActive = this.selectedCombatant.hasTalent(SPELLS.UPWELLING_TALENT.id);
+  }
+
+  calculateMaxDuration(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.ENVELOPING_MIST.id){
+      return (6000 + (this.mistwrapActive ? 1000 : 0)) * 2;
+    }
+    if(spellId === SPELLS.ESSENCE_FONT_BUFF.id){
+      return (8000 + (this.upwellingActive ? 4000 : 0)) * 2;
+    }
+    if(event.selectedCombatant.hasBuff(SPELLS.THUNDER_FOCUS_TEA.id) && spellId === SPELLS.RENEWING_MIST_HEAL.id){
+      return 30000 * 2;
+    }
+    return 20000 * 2;
+  }
+
+  calculateMaxRemDuration(event){
+    const spellId = event.ability.guid;
+    if(event.selectedCombatant.hasBuff(SPELLS.THUNDER_FOCUS_TEA.id) && spellId === SPELLS.RENEWING_MIST_HEAL.id){
+      return 30000;
+    }
+    return 20000;
+  }
```

Comment:
calculateMaxDuration and calculateMaxRemDuration have a lot of hardcoded MS values, maybe they should be moved to consts to follow the rest of the codebase in that regard of not having magic numbers?